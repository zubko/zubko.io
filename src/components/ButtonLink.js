import React from 'react';

import Link from './Link';

export default ({
  backgroundColor,
  borderColor,
  title,
  to,
  children,
  ...otherProps
}) => (
  <Link
    usePlainStyle
    to={to}
    css={[styles.button, { backgroundColor, '--boxShadowColor': borderColor }]}
    {...otherProps}>
    {children}
  </Link>
);

const styles = {
  button: {
    userSelect: 'none',
    padding: '0.3em 0.7em 0.2em',
    color: 'white',
    textDecoration: 'none',
    backgroundImage: 'none',
    textShadow: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: '6px',
    textAlign: 'center',
    verticalAlign: 'center',
    transform: 'translate(0, 0)',
    transition: 'all .2s ease',
    cursor: 'pointer',
    overflow: 'hidden',
    boxShadow: '0px 6px 0px var(--boxShadowColor, #000)',
    '&:hover': {
      color: 'white',
      transform: 'translate(0, -2px)',
      'box-shadow': '0px 8px 0px  var(--boxShadowColor, #000)',
      filter: 'brightness(110%)',
    },
    '&:active': {
      transform: 'translate(0, 3px)',
      'box-shadow': '0px 2px 0px  var(--boxShadowColor, #000)',
    },
  },
};
