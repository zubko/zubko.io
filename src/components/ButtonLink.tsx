/**
 * Component which renders a button to be used as a link
 */

import * as React from "react";
import Link from "./Link";

type Props = {
  backgroundColor: string;
  borderColor: string;
  to: string;
  children?: React.Node;
};

export default ({
  backgroundColor,
  borderColor,
  to,
  children,
  ...otherProps
}: Props) => (
  <Link
    usePlainStyle
    to={to}
    css={[styles.button, { backgroundColor, "--boxShadowColor": borderColor }]}
    {...otherProps}
  >
    {children}
  </Link>
);

const styles = {
  button: {
    userSelect: "none",
    padding: "0.3em 0.7em 0.2em",
    color: "white",
    textDecoration: "none",
    backgroundImage: "none",
    textShadow: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    borderRadius: "6px",
    textAlign: "center",
    verticalAlign: "center",
    transform: "translate(0, 0)",
    transition: "all .2s ease",
    cursor: "pointer",
    overflow: "hidden",
    boxShadow: "0px 6px 0px var(--boxShadowColor, #000)",
    "&:hover": {
      color: "white",
      transform: "translate(0, -2px)",
      boxShadow: "0px 8px 0px  var(--boxShadowColor, #000)",
      filter: "brightness(110%)",
    },
    "&:active": {
      transform: "translate(0, 3px)",
      boxShadow: "0px 2px 0px  var(--boxShadowColor, #000)",
    },
  },
};
