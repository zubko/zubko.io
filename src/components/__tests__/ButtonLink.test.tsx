import React from 'react';
import renderer from 'react-test-renderer';

import ButtonLink from '../ButtonLink';

describe('ButtonLink', () => {
  it('renders without regressions', () => {
    const tree = renderer
      .create(
        <ButtonLink
          backgroundColor="red"
          borderColor="blue"
          to="https://button.link.to">
          Title
        </ButtonLink>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
