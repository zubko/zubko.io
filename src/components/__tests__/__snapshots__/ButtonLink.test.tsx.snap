// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink renders without regressions 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1m4e9d3{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;padding:0.3em 0.7em 0.2em;color:white;-webkit-text-decoration:none;text-decoration:none;background-image:none;text-shadow:none;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;border-radius:6px;text-align:center;vertical-align:center;-webkit-transform:translate(0, 0);-moz-transform:translate(0, 0);-ms-transform:translate(0, 0);transform:translate(0, 0);-webkit-transition:all .2s ease;transition:all .2s ease;cursor:pointer;overflow:hidden;box-shadow:0px 6px 0px var(--boxShadowColor, #000);background-color:red;--boxShadowColor:blue;}.css-1m4e9d3:hover{color:white;-webkit-transform:translate(0, -2px);-moz-transform:translate(0, -2px);-ms-transform:translate(0, -2px);transform:translate(0, -2px);box-shadow:0px 8px 0px var(--boxShadowColor, #000);-webkit-filter:brightness(110%);filter:brightness(110%);}.css-1m4e9d3:active{-webkit-transform:translate(0, 3px);-moz-transform:translate(0, 3px);-ms-transform:translate(0, 3px);transform:translate(0, 3px);box-shadow:0px 2px 0px var(--boxShadowColor, #000);}",
      }
    }
    data-emotion="css 1m4e9d3"
  />,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-7l3lyj{color:#2C8127;-webkit-text-decoration:none;text-decoration:none;background-image:linear-gradient(
    to top,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0) 1px,
    currentColor 1px,
    currentColor 2px,
    rgba(0, 0, 0, 0) 2px
  );-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;padding:0.3em 0.7em 0.2em;color:white;-webkit-text-decoration:none;text-decoration:none;background-image:none;text-shadow:none;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;border-radius:6px;text-align:center;vertical-align:center;-webkit-transform:translate(0, 0);-moz-transform:translate(0, 0);-ms-transform:translate(0, 0);transform:translate(0, 0);-webkit-transition:all .2s ease;transition:all .2s ease;cursor:pointer;overflow:hidden;box-shadow:0px 6px 0px var(--boxShadowColor, #000);background-color:red;--boxShadowColor:blue;}.css-7l3lyj:hover{background-image:none;-webkit-filter:brightness(1.3);filter:brightness(1.3);}.css-7l3lyj:hover{color:white;-webkit-transform:translate(0, -2px);-moz-transform:translate(0, -2px);-ms-transform:translate(0, -2px);transform:translate(0, -2px);box-shadow:0px 8px 0px var(--boxShadowColor, #000);-webkit-filter:brightness(110%);filter:brightness(110%);}.css-7l3lyj:active{-webkit-transform:translate(0, 3px);-moz-transform:translate(0, 3px);-ms-transform:translate(0, 3px);transform:translate(0, 3px);box-shadow:0px 2px 0px var(--boxShadowColor, #000);}",
      }
    }
    data-emotion="css 7l3lyj"
  />,
  <a
    className="css-7l3lyj"
    href="https://button.link.to"
    rel="noopener noreferrer"
    target="_blank"
  >
    Title
  </a>,
]
`;
